src/main/scala/mesosphere/marathon/core/task/termination/impl/TaskKillProgressActor.scala

```diff
@@ -0,0 +1,70 @@
+package mesosphere.marathon.core.task.termination.impl
+
+import akka.actor.{ Actor, ActorLogging, Props }
+import mesosphere.marathon.KillingTasksFailedException
+import mesosphere.marathon.core.task.Task
+import mesosphere.marathon.core.event.MesosStatusUpdateEvent
+
+import scala.concurrent.Promise
+import scala.collection.mutable
+import scala.util.Try
+
+/**
+  * This actor watches over a given set of taskIds and completes a promise when
+  * all tasks have been reported terminal (including LOST et al)
```

Comment:
It also subscribes to the event bus to listen for interesting mesos task status updates (this behavior is deprecated and will be removed in a future release (because MesosStatusUpdateEvent is)?).